Added SQL PowerShell Examples (#1787)
PowerShell/vscode-powershell: docs/azure_data_studio/README_FOR_MARKETPLACE.md

code (Get-ChildItem $Home\.azuredatastudio\extensions\ms-vscode.PowerShell-*\examples)[-1]
```

> ## SQL PowerShell Examples

Reviewer: ```suggestion
### SQL PowerShell Examples
```

Reviewer: So it becomes a sub heading

In order to use these examples (below), you need to install the SqlServer module from the [PowerShell Gallery](https://www.powershellgallery.com/packages/SqlServer).

> ```

Reviewer: ```suggestion
```powershell
```

Install-Module -Name SqlServer -AllowPrerelease
```

In this example, we use the `Get-SqlInstance` cmdlet to Get the Server SMO objects for ServerA & ServerB.  The default output for this command will include the Instance name, version, Service Pack, & CU Update Level of the instances.

> ```

Reviewer: ```suggestion
```powershell
```

Get-SqlInstance -ServerInstance ServerA, ServerB
```

Here is a sample of what that output will look like:

```
Instance Name             Version    ProductLevel UpdateLevel
-------------             -------    ------------ -----------
ServerA                   13.0.5233  SP2          CU4
ServerB                   14.0.3045  RTM          CU12
```

In this example, we will do a `dir` (alias for `Get-ChildItem`) to get the list of all SQL Server instances listed in your Registered Servers file, and then use the `Get-SqlDatabase` cmdlet to get a list of Databases for each of those instances.

> ```

Reviewer: ```suggestion
```powershell
```

dir 'SQLSERVER:\SQLRegistration\Database Engine Server Group' -Recurse |

> WHERE {$_.Mode -ne 'd' } |

Reviewer: ```suggestion
WHERE { $_.Mode -ne 'd' } |
```

FOREACH {

>         Get-SqlDatabase -ServerInstance $_.Name

Reviewer: ```suggestion
    Get-SqlDatabase -ServerInstance $_.Name
```

>         }

Reviewer: ```suggestion
}
```

```

> Here is a sample of what that output will look like:

Reviewer: ```suggestion

Here is a sample of what that output will look like:

```

                                            Available  Model     Level      
----                 ------           ---- ---------- -------- ------- -----
AdventureWorks2017   Normal      336.00 MB   57.01 MB Simple       140 sa   
master               Normal        6.00 MB  368.00 KB Simple       140 sa   
model                Normal       16.00 MB    5.53 MB Full         140 sa   
msdb                 Normal       48.44 MB    1.70 MB Simple       140 sa   
PBIRS                Normal      144.00 MB   55.95 MB Full         140 sa   
PBIRSTempDB          Normal       16.00 MB    4.20 MB Simple       140 sa   
SSISDB               Normal      325.06 MB   26.21 MB Full         140 sa   
tempdb               Normal       72.00 MB   61.25 MB Simple       140 sa   
WideWorldImporters   Normal         3.2 GB     2.6 GB Simple       130 sa   
```

This example uses the `Get-SqlDatabase` cmdlet to retrieve a list of all databases on the ServerB instance, then presents a grid/table (using the `Out-GridView` cmdlet) to select which databases should be backed up.  Once the user clicks on the "OK" button, only the highlighted databases will be backed up.

> ```

Reviewer: ```suggestion
```powershell
```

Get-SqlDatabase -ServerInstance ServerB |
Out-GridView -PassThru |
Backup-SqlDatabase -CompressionOption On
```

This example, again, gets list of all SQL Server instances listed in your Registered Servers file, then calls the `Get-SqlAgentJobHistory` which reports every failed SQL Agent Job since Midnight, for each SQL Server instances listed.

> ```

Reviewer: ```suggestion
```powershell
```

dir 'SQLSERVER:\SQLRegistration\Database Engine Server Group' -Recurse |
WHERE {$_.Mode -ne 'd' } |
FOREACH {

>         Get-SqlAgentJobHistory -ServerInstance  $_.Name -Since Midnight -OutcomesType Failed

Reviewer: ```suggestion
    Get-SqlAgentJobHistory -ServerInstance  $_.Name -Since Midnight -OutcomesType Failed
```

>         }

Reviewer: ```suggestion
}
```

[conduct-FAQ]: http://opensource.microsoft.com/codeofconduct/faq/
[conduct-email]: mailto:opencode@microsoft.com
[conduct-md]: https://github.com/PowerShell/vscode-powershell/blob/master/CODE_OF_CONDUCT.md

## SQL PowerShell Examples
In order to use these examples (below), you need to install the SqlServer module from the [PowerShell Gallery](https://www.powershellgallery.com/packages/SqlServer).

> <pre><code>Install-Module -Name SqlServer -AllowPrerelease</code></pre>

Reviewer 2: Is there a reason we are recommending a prerelease version of this module rather than the last stable version?

Author: Ah, great catch.  I pulled that in from my recent post.  The latest Prerelease includes Invoke-Sqlcmd working on PSCore.  Since Azure Data Studio is cross-platform, having the x-plat Invoke-Sqlcmd is very useful and relevant to those customers.  We can either pull the " -AllowPrerelease" portion now, or we can wait until the final version of the module ships, and pull it then.  

Although, ironically, if PSCore still hasn't shipped then the customer would have to download that preview as well.  (We could include a link to the post on the [PowerShell blog](https://devblogs.microsoft.com/powershell/invoke-sqlcmd-is-now-available-supporting-cross-platform/) as well, to help the x-plat customers)  

In my mind, there is literally no other SQL PowerShell cmdlet as important as Invoke-Sqlcmd.  I can't ever see broadly suggesting customers use a -AllowPrerelease version after this one ships.

I'm fine with leaving " -AllowPrerelease" until the ned of the month, or pulling it now.  Up to you.

Reviewer: I'm fine with `-AllowPrerelease` but please add a comment as to why we're installing the prerelease one.

In this example, we use the Get-SqlInstance cmdlet to Get the Server SMO objects for ServerA & ServerB.  The default output for this command will include the Instance name, version, Service Pack, & CU Update Level of the instances.

<pre><code>Get-SqlInstance -ServerInstance ServerA, ServerB</code></pre>
Here is a sample of what that output will look like:
<pre><code><# Sample Output #>
Instance Name             Version    ProductLevel UpdateLevel
-------------             -------    ------------ -----------
ServerA                   13.0.5233  SP2          CU4
ServerB                   14.0.3045  RTM          CU12</code></pre>

> In this example, we will do a 'dir' (alias for Get-ChildItem) to get the list of all SQL Server instances listed in your Registered Servers file, and then use the Get-SqlDatabase cmdlet to get a list of Databases for each of those instances.

Reviewer: backticks for markdown formatting.

```suggestion
In this example, we will do a `dir` (alias for `Get-ChildItem`) to get the list of all SQL Server instances listed in your Registered Servers file, and then use the `Get-SqlDatabase` cmdlet to get a list of Databases for each of those instances.
```

Author: Sure thing.  I can take care of that right now.

<pre><code>dir 'SQLSERVER:\SQLRegistration\Database Engine Server Group' -Recurse |
WHERE {$_.Mode -ne 'd' } |
FOREACH {
        Get-SqlDatabase -ServerInstance $_.Name
        }
</code></pre>
Here is a sample of what that output will look like:
<pre><code>
Name                 Status           Size     Space  Recovery Compat. Owner
                                            Available  Model     Level      
----                 ------           ---- ---------- -------- ------- -----
AdventureWorks2017   Normal      336.00 MB   57.01 MB Simple       140 sa   
master               Normal        6.00 MB  368.00 KB Simple       140 sa   
model                Normal       16.00 MB    5.53 MB Full         140 sa   
msdb                 Normal       48.44 MB    1.70 MB Simple       140 sa   
PBIRS                Normal      144.00 MB   55.95 MB Full         140 sa   
PBIRSTempDB          Normal       16.00 MB    4.20 MB Simple       140 sa   
SSISDB               Normal      325.06 MB   26.21 MB Full         140 sa   
tempdb               Normal       72.00 MB   61.25 MB Simple       140 sa   
WideWorldImporters   Normal         3.2 GB     2.6 GB Simple       130 sa   
</code></pre>

> This example uses the Get-SqlDatabase cmdlet to retireve a list of all databases on the ServerB instance, then presents a grid/table to select which databases should be backed up.  Once the user clicks on the "OK" button, only the highlighted databases will be backed up.

Reviewer: ```suggestion
This example uses the `Get-SqlDatabase` cmdlet to retrieve a list of all databases on the ServerB instance, then presents a grid/table (using the `Out-GridView` cmdlet) to select which databases should be backed up.  Once the user clicks on the "OK" button, only the highlighted databases will be backed up.
```

Author: Fixed.

<pre><code>Get-SqlDatabase -ServerInstance ServerB |
Out-GridView -PassThru |
Backup-SqlDatabase -CompressionOption On</code></pre>

> This example again gets list of all SQL Server instances listed in your Registered Servers file, then reports every failed SQL Agent Job since Midnight, for each SQL Server instances listed.

Reviewer: ```suggestion
This example, again, gets list of all SQL Server instances listed in your Registered Servers file, then reports every failed SQL Agent Job since Midnight, for each SQL Server instances listed.
```

Author: Fixed.
